go/flexver/main_test.go

```diff
@@ -9,219 +9,181 @@
-	t("0.17.1-beta.1", opLT, "0.17.1-beta.2"),
-	t("1.4.5_01", opEQ, "1.4.5_01+exp-1.17"),
-	t("1.4.5_01", opEQ, "1.4.5_01+exp-1.17-moretext"),
-	t("14w16a", opLT, "18w40b"),
-	t("18w40a", opLT, "18w40b"),
-	t("1.4.5_01+exp-1.17", opLT, "18w40b"),
-	t("13w02a", opLT, "c0.3.0_01"),
-	t("0.6.0-1.18.x", opLT, "0.9.beta-1.18.x"),
-	t("36893488147419103232", opLT, "36893488147419103233"),
-	t("1.0", opLT, "1.1"),
-	t("1.0", opLT, "1.0.1"),
-	t("10", opGT, "2"),
-}
+	res2, err := LessError(lefthand, righthand)
+	if err != nil {
+		t.Fatalf("LessError returned an unexpected error: %v", err)
+	}
 
-var miscTests = []testInstance{
-	// Empty strings
-	t("", opEQ, ""),
-	t("1", opGT, ""),
-	t("", opLT, "1"),
-	// Invalid UTF8
-	t("\xc3\x28", shouldError, ""),
-	t("", shouldError, "\xc3\x28"),
-	// Check boundary between textual and prerelease
-	t("a-a", opLT, "a"),
-	// Check boundary between textual and appendix
-	t("a+a", opEQ, "a"),
-	// Dash is included in prerelease comparison (if stripped it will be a smaller component)
-	// Note that a-a < a=a regardless since the prerelease splits the component creating a smaller first component; 0 is added to force splitting regardless
-	t("a0-a", opLT, "a0=a"),
-	// Pre-releases must contain only non-digit
-	t("1.16.5-10", opGT, "1.16.5"),
-	// Pre-releases can have multiple dashes (should not be split)
-	// Reasoning for test data: "p-a!" > "p-a-" (correct); "p-a!" < "p-a t-" (what happens if every dash creates a new component)
-	t("-a-", opGT, "-a!"),
+	if res != res2 {
+		t.Error("LessError did not give the same result as Less")
+	}
 }
 
-var allTests = append(append([]testInstance{}, specTests...), miscTests...)
+func RunEqual(t *testing.T, lefthand string, righthand string, ordening int) {
+	res := Equal(lefthand, righthand)
+	if ordening == opEQ {
+		if !res {
+			t.Error("Equal incorrectly returned false")
+		}
+	} else {
+		if res {
+			t.Error("Equal incorrectly returned true")
+		}
+	}
 
-func TestCompare(t *testing.T) {
-	for _, v := range allTests {
-		t.Run(v.name(), func(t *testing.T) {
-			if v.op == shouldError {
-				// Catch panic
-				defer func() { _ = recover() }()
-			}
+	res2, err := EqualError(lefthand, righthand)
+	if err != nil {
+		t.Fatalf("EqualError returned an unexpected error: %v", err)
+	}
 
-			res := Compare(v.a, v.b)
-			if (v.op == opLT && res >= 0) ||
-				(v.op == opEQ && res != 0) ||
-				(v.op == opGT && res <= 0) ||
-				(v.op == shouldError) {
-				t.Fatalf("Unexpected result %v", res)
-			}
-		})
+	if res != res2 {
+		t.Error("EqualError did not give the same result as Equal")
 	}
 }
 
-func TestLess(t *testing.T) {
-	for _, v := range allTests {
-		t.Run(v.name(), func(t *testing.T) {
-			if v.op == shouldError {
-				// Catch panic
-				defer func() { _ = recover() }()
-			}
+func TestStandardized(t *testing.T) {
+	for _, test := range ENABLED_TESTS {
+		file, err := os.Open("../../test/" + test)
+		if err != nil {
+			t.Fatal(err)
+		}
+		defer file.Close()
```

Comment:
This won't close each individual test data file until the whole function returns; the for loop contents could be refactored out into another function (or inline closure)